### PR TITLE
Fix petsc/copy_to_dealvec_block

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes https://cdash.43-1.org/testDetails.php?test=40761410&build=5683.

### CODE REVIEW

**Comment by masterleinad** [comment]
Merging since `appveyor` succeeded.

File: include/deal.II/lac/la_parallel_block_vector.templates.h [hunk 1]
```diff
@@ -221,14 +253,34 @@ namespace LinearAlgebra
       AssertDimension(this->n_blocks(), petsc_vec.n_blocks());
       for (unsigned int i = 0; i < this->n_blocks(); ++i)
         {
-          const auto &partitioner  = this->block(i).get_partitioner();
-          IndexSet    combined_set = partitioner->locally_owned_range();
-          combined_set.add_indices(partitioner->ghost_indices());
-          ReadWriteVector<Number> rw_vector(combined_set);
-          rw_vector.import(petsc_vec.block(i), VectorOperation::insert);
```
Comment (tjhei) [inline include/deal.II/lac/la_parallel_block_vector.templates.h:228]: does this mean that using a RWV silently gives the wrong results without triggering any errors?

Comment (masterleinad) [reply include/deal.II/lac/la_parallel_block_vector.templates.h:228]: No, it was complaining in the failing test linked.
And for clarification, the code was copied from what we previously had in the `operator=(PETScWrapper::MPI::Vector)` overload for the non-block vector.

Comment (tjhei) [reply include/deal.II/lac/la_parallel_block_vector.templates.h:228]: sorry, I happened to click on the release build which does not show the assert. :+1:

**Pull request merged**